src/reactviews/pages/TableDesigner/designerTable.tsx

```diff
@@ -379,97 +343,170 @@ export const DesignerTable = ({
+                                        }
+                                        state?.provider.setPropertiesComponents(
+                                            {
+                                                componentPath: [
+                                                    ...componentPath,
+                                                    row.rowId,
+                                                ],
+                                                component: component,
+                                                model: model,
+                                            },
+                                        );
+                                        setFocusedRowId(index);
+                                        event.preventDefault();
+                                    }}
+                                    key={componentPath.join(".") + index}
+                                    onDragEnter={() => {
+                                        setDraggedOverRowId(index);
+                                    }}
+                                    onDragEnd={() => {
+                                        if (
+                                            draggedRowId === undefined ||
+                                            draggedOverRowId === undefined
+                                        ) {
+                                            return;
+                                        }
+                                        moveRows(
+                                            draggedRowId,
+                                            draggedOverRowId,
+                                        );
+                                        setDraggedRowId(undefined);
+                                        setDraggedOverRowId(undefined);
+                                    }}
+                                    onDrag={() => {
+                                        setDraggedRowId(index);
+                                    }}
+                                    onDragStart={() => {
+                                        setDraggedOverRowId(undefined);
+                                        setDraggedRowId(index);
+                                    }}
                                 >
-                                    {column.renderHeaderCell()}
-                                </fluentui.TableHeaderCell>
+                                    {columnsDef.map((column, columnIndex) => {
+                                        const borderColor =
+                                            state?.provider.getErrorMessage([
+                                                ...componentPath,
+                                                row.rowId,
+                                                column.columnId,
+                                            ])
+                                                ? "1px solid var(--vscode-errorForeground)"
```

Comment:
Is this where the indication for an error on the table is?  A few design thoughts on that front:

* To make it look a little nicer, maybe it can have a slight border radius.
* Since the error border takes up 1px of space, maybe there should be a 1px invisible border if there's no error so that everything stays aligned, or the error border should be an inner border, or the non-error cells get an extra pixel of margin
* or maybe better yet: since the designerXYZ components are fields, set those fields into error states and let the fluent library handle rendering the error border natively for each control?

Reply:
![image](https://github.com/user-attachments/assets/84a5eb75-6c0d-4236-8448-4e537d64e016)
![image](https://github.com/user-attachments/assets/b0cb364f-438e-47a0-bd9a-6a704a7e4420)

Updated styling